### PR TITLE
Refactor wallet details screen

### PR DESCRIPTION
**Pull request by KedziaPawel**

### CODE REVIEW

File: class/app-storage.js [hunk 1]
```diff
@@ -269,6 +269,21 @@ export class AppStorage {
     this.wallets = [...this.wallets, w];
   }
 
+  updateWallet(w) {
```
Comment (Kamill90) [inline class/app-storage.js:272]: what is 'w'? Please, consider rewriting it to ts

File: src/components/WalletCard.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class WalletCard extends React.Component<Props> {
               {showEditButton && (
                 <StyledText
                   title={i18n.wallets.details.edit}
-                  onPress={() => NavigationService.navigate(Route.WalletDetails, { wallet })}
+                  onPress={() => NavigationService.navigate(Route.WalletDetails, { id: wallet.id })}
```
Comment (Kamill90) [inline src/components/WalletCard.tsx:40]: Please move to the class method.

File: src/screens/WalletDetailsScreen.tsx [hunk 3]
```diff
@@ -82,35 +78,47 @@ export class WalletDetailsScreen extends React.PureComponent<Props, State> {
...
+    updatedWalelt.setLabel(trimmedlabel);
+    updateWallet(updatedWalelt);
   };
 
   editAmount = () => {
-    this.props.navigation.navigate(Route.EditText, {
+    const { wallet, navigation } = this.props;
+    if (!wallet) {
+      return;
+    }
```
Comment (Kamill90) [inline src/screens/WalletDetailsScreen.tsx:106]: if there is no wallet in the props user will not see this screen according to the statement in the render method

File: class/app-storage.js (hunk 1 above)
Comment (KedziaPawel) [reply class/app-storage.js:272]: w is the wallet I can rename it to walletUpdate. There is a  separate task for rewriting app-storage to ts

File: src/screens/WalletDetailsScreen.tsx (hunk 3 above)
Comment (KedziaPawel) [reply src/screens/WalletDetailsScreen.tsx:106]: yes, but the TS is complaining without this condition, it is not that smart :D

**Review by Kamill90 — APPROVED** [review]

**Pull request merged**